pandas_gbq/gbq.py

```diff
@@ -956,61 +803,148 @@ def read_gbq(query, project_id=None, index_col=None, col_order=None,
 
     """
 
-    _test_google_api_imports()
-
-    if not project_id:
-        raise TypeError("Missing required parameter: project_id")
-
     if dialect not in ('legacy', 'standard'):
         raise ValueError("'{0}' is not valid for dialect".format(dialect))
+    if configuration and any(key in configuration for key in
+                             ["query", "copy", "load", "extract"]):
+        raise ValueError("New API handles configuration settings differently")
+
+    def _wait_for_job(job):
+        while True:
+            job.reload()  # Refreshes the state via a GET request.
+            if job.state == 'DONE':
+                if job.error_result:
+                    raise RuntimeError(job.errors)
+                return
+            time.sleep(1)
 
-    connector = GbqConnector(
-        project_id, reauth=reauth, verbose=verbose, private_key=private_key,
-        dialect=dialect, auth_local_webserver=auth_local_webserver)
-    schema, pages = connector.run_query(query, **kwargs)
-    dataframe_list = []
-    while len(pages) > 0:
-        page = pages.pop()
-        dataframe_list.append(_parse_data(schema, page))
-
-    if len(dataframe_list) > 0:
-        final_df = concat(dataframe_list, ignore_index=True)
+    if credentials is None:
+        credentials = GbqConnector(project_id=project_id,
+                                   reauth=reauth,
+                                   auth_local_webserver=auth_local_webserver,
+                                   private_key=private_key).credentials
+    client = bigquery.Client(project=project_id, credentials=credentials)
+
+    def _set_common_query_settings(query_job):
+        if dialect == 'legacy':
+            query_job.use_legacy_sql = True
+        elif dialect == 'standard':
+            query_job.use_legacy_sql = False
+
+        if configuration:
+            for setting, value in configuration.items():
+                setattr(query_job, setting, value)
+        return query_job
+
+    def sync_query():
+        query_job = client.run_sync_query(query,
+                                          query_parameters=query_parameters)
+        query_job = _set_common_query_settings(query_job)
+        if verbose:
+            print("Query running...")
```

Comment:
since you are re-writing. I would remove all prints and simply use the standard logging module, much more modular / extensible.

Reply:
Can we leave that for another day since I don't know the convention and there's no model to follow here?

Reply:
yes this is ok